ssh.sh

```diff
@@ -1,121 +1,291 @@
 #!/bin/bash
 
-cd $HOME
-mkdir -p .ssh
-cd .ssh
+# Exit on any error
+set -e
 
-echo "Select an option:"
-echo "1. Bitbucket"
-echo "2. Github"
-echo "3. GitLab"
-read -p "Enter your choice: " option
+# Function to detect operating system
+detect_os() {
+    if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+        if grep -q Microsoft /proc/version 2>/dev/null; then
+            echo "wsl"
+        else
+            echo "linux"
+        fi
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        echo "macos"
+    elif [[ "$OSTYPE" == "cygwin" ]] || [[ "$OSTYPE" == "msys" ]] || [[ "$OSTYPE" == "win32" ]]; then
+        echo "windows"
+    else
+        echo "unknown"
+    fi
+}
+
+# Navigate to SSH directory with error handling
+cd "$HOME" || { echo "Error: Cannot access home directory"; exit 1; }
+mkdir -p .ssh || { echo "Error: Cannot create .ssh directory"; exit 1; }
+cd .ssh || { echo "Error: Cannot access .ssh directory"; exit 1; }
 
-echo "Enter your email: "
-read email
-echo "Enter your SSH key name: "
-read keyName
+# Get user input with validation
+while true; do
+    echo "Select an option:"
+    echo "1. Bitbucket"
+    echo "2. Github"
+    echo "3. GitLab"
+    read -r -p "Enter your choice (1-3): " option
+    
+    if [[ "$option" =~ ^[1-3]$ ]]; then
```

Comment:
This Rehex doesn't seem to work on my Machine : 
`ssh.sh: 36: [[: Permission denied
Error: Please enter a valid option (1, 2, or 3)
Select an option:
1. Bitbucket
2. Github
3. GitLab
Enter your choice (1-3):`